src/cpu/o3/trace/ChampSimTraceReader.test.cc

```
@@ -0,0 +1,525 @@
+        if ((mapped - base) >= size) {
+            mapped = base + (trace_addr % size);
+        }
+    }
+    return mapped;
+}
+
+static inline uint64_t
+mapLinear(uint64_t trace_addr, uint64_t base, uint64_t size, bool page_align)
+{
+    constexpr uint64_t PAGE_SIZE = 4096ULL;
+    uint64_t mapped;
```

Cppcheck (2.18.0)

[error] Shifting 64-bit value by 64 bits is undefined behaviour

(shiftTooManyBits)